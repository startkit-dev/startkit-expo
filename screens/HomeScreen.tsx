import { Pressable, Text, View } from "@/components/elements"

import type { RootStackScreenProps } from "@/navigation/root/types"

export function HomeScreen({ navigation }: RootStackScreenProps<"Home">) {
  return (
    <>
      <View className="flex-1 items-center p-6">
        <View className="mx-auto flex-1 justify-center">
          <Text className="text-6xl font-bold">Hello World</Text>
          <Text className="text-4xl text-gray-800 dark:text-gray-100">
            This is the first page of your app.
          </Text>
          <Text className="mt-2 text-xl">
            Open `<Text mono>screens/HomeScreen.tsx</Text>` to get started.
          </Text>

          <Pressable
            onPress={() => navigation.navigate("Settings")}
            className="mx-auto my-8"
          >
            <Text>Open Modal</Text>
          </Pressable>
        </View>
      </View>
    </>
  )
}
