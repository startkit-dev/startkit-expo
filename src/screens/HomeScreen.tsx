import { StyleSheet, Text, View } from "react-native";

export default function HomeScreen() {
  return (
    <View style={styles.container}>
      <Text>
        Open up src/screens/HomeScreen.tsx to start working on your app!
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
