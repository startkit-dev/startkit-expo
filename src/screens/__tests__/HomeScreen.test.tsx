import { create } from "react-test-renderer";

import HomeScreen from "@/screens/HomeScreen";

test("renders correctly", () => {
  const tree = create(<HomeScreen />).toJSON();

  expect(tree).toMatchSnapshot();
});
