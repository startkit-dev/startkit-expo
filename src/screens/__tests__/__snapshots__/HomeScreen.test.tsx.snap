// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text>
    Open up src/screens/HomeScreen.tsx to start working on your app!
  </Text>
</View>
`;
